crates/unftp-auth-jsonfile/src/lib.rs

```diff
@@ -101,14 +133,31 @@ impl JsonFileAuthenticator {
 impl Authenticator<DefaultUser> for JsonFileAuthenticator {
     #[allow(clippy::type_complexity)]
     #[tracing_attributes::instrument]
-    async fn authenticate(&self, username: &str, password: &str) -> Result<DefaultUser, AuthenticationError> {
+    async fn authenticate(&self, username: &str, pass: &str) -> Result<DefaultUser, AuthenticationError> {
         let db: HashMap<String, Password> = self.db.clone();
 
         if let Some(c) = db.get(username) {
-            if let Ok(()) = verify(PBKDF2_HMAC_SHA512, c.pbkdf2_iter, c.pbkdf2_salt.as_bytes(), password.as_bytes(), &c.pbkdf2_key) {
+            if let Ok(()) = match c {
```

Comment:
I suggest refactoring the match statement into its own function e.g. `valid_credentials`